src/node_realm.cc

```diff
@@ -359,4 +304,88 @@ void Realm::VerifyNoStrongBaseObjects() {
   });
 }
 
+v8::Local<v8::Context> Realm::context() const {
+  return PersistentToLocal::Strong(context_);
+}
+
+#define V(PropertyName, TypeName)                                              \
+  v8::Local<TypeName> PrincipalRealm::PropertyName() const {                   \
+    return PersistentToLocal::Strong(PropertyName##_);                         \
+  }                                                                            \
+  void PrincipalRealm::set_##PropertyName(v8::Local<TypeName> value) {         \
+    PropertyName##_.Reset(isolate(), value);                                   \
+  }
+PER_REALM_STRONG_PERSISTENT_VALUES(V)
```

Comment:
After a glance at `PER_REALM_STRONG_PERSISTENT_VALUES` I wonder if we should just call them `PRINCIPAL_REALM_STRONG_PERSISTENT_VALUES` or `NODE_REALM_STRONG_PERSISTENT_VALUES `, because not all of them appear to be necessary for shadow realms to me..